contracts/ProfileToken.sol

```diff
@@ -13,25 +13,62 @@ contract ProfileToken is
 {
     using Counters for Counters.Counter;
 
+    // data
+
     Counters.Counter public _tokenIds;
 
+    struct Avatar {
+        address nftContract;
+        uint256 tokenId;
+    }
+    Avatar private _avatarInfo;
+
+    // events
+
     event NewProfile(uint256 tokenId, address owner);
+    event SetAvatar(uint256 indexed tokenId, address avatarContract, uint256 avatarId, string avatarURI);
+
+    // functions
 
     constructor() {
     }
 
     // contract owner can mint to anyone, anyone can mint to themself, unlimited supply
-    function mint(address to) public returns (uint256) {
+    function mint(address to) external returns (uint256) {
         require(msg.sender == owner() || msg.sender == to, "mint not allowed");
 
         uint256 newProfileId = _tokenIds.current();
         _safeMint(to, newProfileId);
 
-        // console.log("Minted profile: %s to %s", newProfileId, to);
+        //console.log("Minted profile: %s", newProfileId);
         emit NewProfile(newProfileId, to);
 
         _tokenIds.increment();
 
         return newProfileId;
     }
+
+    // associate an avatar with a profile. must be owner of both tokens.
+    function setAvatar(address avatarContractAddrs, uint avatarTokenId, uint niftyTokenId) public {
+        require(msg.sender == ERC721.ownerOf(niftyTokenId), "not profile owner");
+
+        ERC721 avatarNFT = ERC721(avatarContractAddrs);
+        require(msg.sender == avatarNFT.ownerOf(avatarTokenId), "not avatar owner");
+
+        _avatarInfo.nftContract = avatarContractAddrs;
+        _avatarInfo.tokenId = avatarTokenId;
+
+        emit SetAvatar(niftyTokenId, avatarContractAddrs, avatarTokenId, tokenURI(niftyTokenId));
```

Comment:
What is the `tokenURI(niftyTokenId)` doing? I think at this point, the result of this is just an empty string. I may be missing something, so can you please explain?